examples/speaker_diarization/speaker_diarization/seed_test_run.py

```diff
@@ -0,0 +1,114 @@
+
+import pandas as pd
+from evaluator import evaluate_speaker_diarization
+from utils import realign_labels
+from workflow import Inference
+from workflow import Model
+from workflow import TestSample
+from workflow import TestSuite
+
+import kolena
+from kolena.workflow.annotation import LabeledTimeSegment
+from kolena.workflow.test_run import test
+
+BUCKET = "kolena-public-datasets"
+DATASET = "ICSI-corpus"
+
+TEST_SUITE_NAMES = [
+    f"{DATASET} :: average amplitude",
+]
+
+MODEL_A = {
+    "model family": "GCP Speech To Text",
+    "model name": "gcp-stt-video",
+    "description": "Speaker diarization model optimized for videos.",
+}
+
+MODEL_METADATA: Dict[str, Dict[str, str]] = {
+    "gcp-stt-video": MODEL_A,
+}
+
+
+def seed_test_run(
+    mod: str,
+    test_suite: TestSuite,
+    align_speakers: bool,
+) -> None:
+    def infer(sample: TestSample) -> Inference:
+        inference_path = sample.metadata["transcription_path"].replace("audio/", f"{mod}_inferences/")
+        inference_df = pd.read_csv(f"s3://{BUCKET}/{DATASET}/{inference_path}/")
+        if align_speakers:
+            gt_df = pd.read_csv(
+                f"s3://{BUCKET}/{DATASET}/{sample.metadata['transcription_path'][:-4] + '_cleaned.csv'}",
+            )
+            realign_labels(gt_df, inference_df)
+
+        return Inference(
+            transcription=[
+                LabeledTimeSegment(
+                    start=row.starttime,
+                    end=row.endtime,
+                    label=row.text,
+                    group=row.speaker,
+                )
+                for _, row in inference_df.iterrows()
+            ],
+        )
+
+    print(f"working on {mod} and {test_suite.name} v{test_suite.version}")
+    model = Model(f"{mod}", infer=infer, metadata={**MODEL_METADATA[mod]})
+
+    test(model, test_suite, evaluate_speaker_diarization, reset=True)
+
+
+def main(args: Namespace) -> None:
+    kolena.initialize(verbose=True)
+
+    mod = args.model
+
+    if args.test_suite is None:
+        print("loading test suite")
+        test_suites = [TestSuite.load(name) for name in TEST_SUITE_NAMES]
+        for test_suite in test_suites:
+            seed_test_run(mod, test_suite, args.align_speakers)
+    else:
+        test_suite = TestSuite.load(args.test_suite)
+        seed_test_run(mod, test_suite, args.align_speakers)
+
+
+if __name__ == "__main__":
+    ap = ArgumentParser()
+    ap.add_argument(
+        "model",
+        type=str,
+        choices=sorted(MODEL_METADATA.keys()),
+        help="The name of the model to test.",
+    )
+    ap.add_argument(
+        "--test-suite",
+        type=str,
+        help="Optionally specify a test suite to test. Test against all available test suites when unspecified.",
+    )
+    ap.add_argument(
+        "--align-speakers",
+        type=bool,
+        default=False,
+        help="Specify whether to perform speaker alignment between the GT and Inf in the preprocessing step.",
+    )
```

Comment:
When would we ever not want to do this?

Reply:
I previously had it align the speakers in the inference step, but later moved the alignment into the CSVs directly to save some time. However, I wanted to keep this in case they wanted to upload their own data using this workflow